src/provider/systemdmanager/systemdunit.rs

```diff
@@ -358,3 +459,156 @@ impl Display for SystemDUnit {
+                  restartPolicy: Always
+                  securityContext:
+                    windowsOptions:
+                      runAsUserName: pod-user"},
+            "stackable.service",
+            indoc! {"
+                [Service]
+                Restart=always
+                User=pod-user"}
+        ),
+        case::with_container_on_system_bus(
+            BusType::System,
+            indoc! {r#"
+                apiVersion: v1
+                kind: Pod
+                metadata:
+                  name: stackable
+                spec:
+                  containers:
+                    - name: test-container.service
+                      command:
+                        - start.sh
+                      args:
+                        - arg
+                        - "{{configroot}}"
+                      env:
+                        - name: LOG_LEVEL
+                          value: INFO
+                        - name: LOG_DIR
+                          value: "{{logroot}}"
+                      securityContext:
+                        windowsOptions:
+                          runAsUserName: container-user
+                  securityContext:
+                    windowsOptions:
+                      runAsUserName: pod-user"#},
+            "default-stackable-test-container.service",
+            indoc! {r#"
+                [Unit]
+                Description=default-stackable-test-container
+
+                [Service]
+                Environment="LOG_DIR=/var/log/default-stackable" "LOG_LEVEL=INFO"
+                ExecStart=start.sh arg /etc/default-stackable
+                Restart=no
+                StandardError=journal
+                StandardOutput=journal
+                User=container-user
+
+                [Install]
+                WantedBy=multi-user.target"#}
+        ),
+        case::with_container_on_session_bus(
+            BusType::Session,
+            indoc! {r#"
+                apiVersion: v1
+                kind: Pod
+                metadata:
+                  name: stackable
+                spec:
+                  containers:
+                    - name: test-container.service
+                      command:
+                        - start.sh
+                      securityContext:
+                        windowsOptions:
+                          runAsUserName: container-user
+                  securityContext:
+                    windowsOptions:
+                      runAsUserName: pod-user"#},
+            "default-stackable-test-container.service",
+            indoc! {r#"
+                [Unit]
+                Description=default-stackable-test-container
+
+                [Service]
+                ExecStart=start.sh
+                Restart=no
+                StandardError=journal
+                StandardOutput=journal
+
+                [Install]
+                WantedBy=multi-user.target"#}
+        ),
+    )]
+    fn create_unit_from_pod(
```

Comment:
This is fabulous and I'm fine with merging as is but it's not testing any of the error cases like a wrong user name

Reply:
It is correct that nearly every line on the happy path is tested but the exception path is ignored. It is definitely worth to test the error cases, too. But my goal is to parse and validate the PodSpec in a first step and then pass this valid structure around where further validation is not necessary. This keeps the tests in the downstream modules comprehensible. Otherwise it would be necessary to define a full PodSpec to test the regular expression for user names and another full PodSpec to test the order of the environment variables and so on. If everything is tested where it belongs to then the tests can be reduced to 3-liners.